File: saas-admin/src/types/activityLog.ts
import { Status } from "./app";

export type ActivityLogDto = {
  id: string;
  status: Status;
  createdDate: string;
  modifiedDate: string;
  autoID: number;
  userId: string | null;
  userName: string | null;
  userEmail: string | null;
  appId: string | null;
  appName: string | null;
  appCode: string | null;
  action: string;
  entityType: string;
  entityId: string | null;
  description: string;
  oldValues: string | null;
  newValues: string | null;
  ipAddress: string | null;
  userAgent: string | null;
  requestId: string | null;
};

File: SaasV2.Entity/ActivityLog.cs
using System;

namespace SaasV2.Entity
{
    /// <summary>
    /// Sistem aktivite logları - kim ne yaptı, ne zaman yaptı
    /// </summary>
    public class ActivityLog : BaseEntity
    {
        public Guid? UserId { get; set; }           // Kim yaptı (nullable - sistem işlemleri için)
        public Guid? AppId { get; set; }            // Hangi uygulama için (nullable)
        
        public string Action { get; set; }           // Create, Update, Delete, Cancel, Activate, etc.
        public string EntityType { get; set; }       // Subscription, Invoice, Plan, App, User, etc.
        public Guid? EntityId { get; set; }         // Hangi kayıt (nullable)
        
        public string Description { get; set; }      // İnsan okunabilir açıklama
        public string? OldValues { get; set; }       // Eski değerler (JSON, nullable)
        public string? NewValues { get; set; }       // Yeni değerler (JSON, nullable)
        
        public string? IpAddress { get; set; }       // IP adresi (nullable)
        public string? UserAgent { get; set; }       // User agent (nullable)
        public string? RequestId { get; set; }       // Request correlation ID (nullable)
    }
}

